src/main/java/com/nike/cerberus/aws/sts/AwsStsHttpClient.java

```diff
@@ -0,0 +1,147 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.aws.sts;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nike.backstopper.exception.ApiException;
+import com.nike.cerberus.error.DefaultApiError;
+import com.nike.cerberus.server.config.guice.AwsStsGuiceModule;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.net.ssl.SSLException;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * A HttpClient for interacting with AWS STS
+ */
+@Singleton
+public class AwsStsHttpClient {
+
+    private static final MediaType DEFAULT_CONTENT_MEDIA_TYPE = MediaType.parse("application/x-www-form-urlencoded");
+
+    private static final MediaType DEFAULT_ACCEPTED_MEDIA_TYPE = MediaType.parse("application/json");
+
+    private final String DEFAULT_AWS_STS_ENDPOINT = "https://sts.amazonaws.com";
+
+    private final String DEFAULT_GET_CALLER_IDENTITY_ACTION = "Action=GetCallerIdentity&Version=2011-06-15";
+
+    private final String DEFAULT_METHOD = "POST";
+
+    private final OkHttpClient httpClient;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public AwsStsHttpClient(@Named(AwsStsGuiceModule.AWS_STS_HTTP_CLIENT_NAME) final OkHttpClient httpClient,
+                            @Named(AwsStsGuiceModule.AWS_STS_OBJECT_MAPPER_NAME) final ObjectMapper objectMapper) {
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Executes the HTTP request based on the input parameters.
+     *
+     * @param headers     HTTP Headers to include in the request
+     * @param responseClass The class of the response object
+     * @return Response from the server
+     */
+    public <M> M execute(final Map<String, String> headers,
+                            final Class<M> responseClass) {
+        try {
+            Request request = buildRequest(headers);
+            Response response =  httpClient.newCall(request).execute();
+            if (!response.isSuccessful()) {
+                final String msg = String.format("Failed to authenticate with AWS, error message: %s",
+                        response.body().string());
+
+                throw ApiException.newBuilder()
```

Comment:
You are throwing this error assuming what a 403/401?

There are many other 400-500 series things that can go wrong, always throwing AUTH_BAD_CREDENTIALS even if a rate limit is hit or a service is down seems wrong.